src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Decompression.cs

```diff
@@ -184,10 +193,16 @@ public async Task GetAsync_SetAutomaticDecompression_ContentDecompressed(Configu
         [Theory, MemberData(nameof(RemoteServersAndCompressionUris))]
         public async Task GetAsync_SetAutomaticDecompression_HeadersRemoved(Configuration.Http.RemoteServer remoteServer, Uri uri)
         {
+            // Sync API supported only up to HTTP/1.1
+            if (!TestAsync && remoteServer.HttpVersion.Major >= 2)
+            {
+                return;
+            }
+
             HttpClientHandler handler = CreateHttpClientHandler();
             handler.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
             using (HttpClient client = CreateHttpClientForRemoteServer(remoteServer, handler))
-            using (HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead))
+            using (HttpResponseMessage response = await client.SendAsync(TestAsync, CreateRequest(HttpMethod.Get, uri, remoteServer.HttpVersion), HttpCompletionOption.ResponseHeadersRead))
```

Comment:
`SendAsync|Send` ignores `HttpClient.DefaultRequestVersion` since it accepts fully constructed request. The default initialization of [`HttpRequestMessage`](https://github.com/dotnet/runtime/blob/d927afce85f66f80e82a7a74adf01ee3692caa1a/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs#L194) will set it to [1.1](https://github.com/dotnet/runtime/blob/d927afce85f66f80e82a7a74adf01ee3692caa1a/src/libraries/System.Net.Http/src/System/Net/Http/HttpUtilities.cs#L12). And because we plug in [`VersionChecker`](https://github.com/dotnet/runtime/blob/d927afce85f66f80e82a7a74adf01ee3692caa1a/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs#L92) set up with the remote server version, it will blow up for 2.0 servers.